src/dtls/mod.rs

```diff
@@ -48,6 +48,43 @@ pub struct Fingerprint {
     pub bytes: Vec<u8>,
 }
 
+impl ToString for Fingerprint {
+    /// Convert to the hex string you find in SDP
+    fn to_string(&self) -> String {
+        format!(
+            "{} {}",
+            self.hash_func,
+            self.bytes
+                .iter()
+                .map(|b| format!("{:02X}", b))
+                .collect::<Vec<_>>()
+                .join(":")
+        )
+    }
+}
```

Comment:
This is actually doubling up on functionality we have. For some reason, and it should probably be cleaned up sometime. The types that are serialized to SDP does so via the `Display` trait. Very wrong. 

I don't mind having it as `ToString` too, but let's just delegate to the one impl in that case. The `Display` variant is a bit more efficient, not requiring an intermediate Vec.

Reply:
ok, so maybe break out some of the code from impl<'a> std::fmt::Display for FingerprintFmt<'a> into a separate function and then use it from both places?